File: AgenticChatbot/models/customer.py
from pydantic import BaseModel
from typing import List
from datetime import datetime

class Address(BaseModel):
    addressLine1: str
    addressLine2: str
    city: str
    state: str
    country: str
    postalCode: str

class PhoneInfo(BaseModel):
    homePhone: str
    workPhone: str

class Customer(BaseModel):
    customerId: str
    ssn: str
    customerName: str
    createdOn: datetime
    address: Address
    emailAddress: str
    phoneInfo: PhoneInfo
    paymentReminder: bool

File: AgenticAPI/AgenticAPI/Domain/Customer.cs
using MongoDB.Bson.Serialization.Attributes;

namespace AgenticAPI.Domain
{
    [BsonIgnoreExtraElements]
    public class Customer
    {
        public string? CustomerId { get; set; }
        public string? SSN { get; set; }
        public string? CustomerName { get; set; }
        public DateTime? CreatedOn { get; set; }
        public Address? Address { get; set; }
        public string? EmailAddress { get; set; }
        public PhoneInfo? PhoneInfo { get; set; }
        public bool? PaymentReminder { get; set; }
    }
}
